### PR TITLE
single sample fast mode for qual is HC isActive

### LINKED ISSUE #6111: Last relic of old qual
`HaplotypeCallerEngine::isActive` still invokes a `calculateSingleSampleRefVsAnyActiveStateProfileValue` method that relies on allele count priors and dodgy math left over from the old days.  Can we use `AlleleFrequencyCalculator` instead?

### PR DESCRIPTION
**Pull request by davidbenjamin**
Closes #6111.

The qual used in HC's `isActive` method now matches that of `GenotypeGVCFs`.  That is, they both now use the new qual.  This lets us finally delete `AFPriorProvider`.  I have profiled and HaplotypeCaller speeds up in this branch by a negligible but measurable amount.

@ldgauthier Can you look at this?  the big paragraph changed in the new qual notes is just correcting the omission of some `-1`s and has nothing to do with the single-sample calculation.

### CODE REVIEW

File: src/main/java/org/broadinstitute/hellbender/tools/walkers/genotyper/GenotypeCalculationArgumentCollection.java [hunk 1]
```diff
@@ -152,29 +151,6 @@ public GenotypeCalculationArgumentCollection( final GenotypeCalculationArgumentC
...
-     * d) No negative values allowed.
-     * e) Values will be added and Pr(AC=0) will be 1-sum, so that they sum up to one.
-     * f) If user-defined values add to more than one, an error will be produced.
-     *
-     * If user wants completely flat priors, then user should specify the same value (=1/(2*N+1)) 2*N times,e.g.
-     *   --input-prior 0.33 --input-prior 0.33
-     * for the single-sample diploid case.
-     */
-    @Advanced
-    @Argument(fullName = "input-prior",  doc = "Input prior for calls", optional = true)
```
Comment (ldgauthier) [inline src/main/java/org/broadinstitute/hellbender/tools/walkers/genotyper/GenotypeCalculationArgumentCollection.java]: Oh thank goodness this is gone.

File: src/main/java/org/broadinstitute/hellbender/tools/walkers/genotyper/afcalc/AlleleFrequencyCalculator.java [hunk 2]
```diff
@@ -154,6 +155,30 @@ public AFCalculationResult calculate(final VariantContext vc, final int defaultP
         return new AFCalculationResult(integerAltAlleleCounts, alleles, log10PNoVariant, log10PRefByAllele);
     }
 
+    /**
+     * Calculate the posterior probability that a single biallelic sample is non-ref
```
Comment (ldgauthier) [inline src/main/java/org/broadinstitute/hellbender/tools/walkers/genotyper/afcalc/AlleleFrequencyCalculator.java]: I don't like the phrase `biallelic sample`, but I get that the gist is single-sample.  Maybe biallelic genotype?

File: src/test/java/org/broadinstitute/hellbender/tools/walkers/genotyper/afcalc/AlleleFrequencyCalculatorUnitTest.java [hunk 3]
```diff
@@ -292,6 +293,36 @@ public void testSpanningDeletionWithVeryUnlikelyAltAllele() {
         final double log10PVariant = afCalc.calculate(vc).log10ProbVariantPresent();
     }
 
+    //check the exact integration calculation
+    @Test
```
Comment (ldgauthier) [inline src/test/java/org/broadinstitute/hellbender/tools/walkers/genotyper/afcalc/AlleleFrequencyCalculatorUnitTest.java]: Lovely tests, as always.

File: src/main/java/org/broadinstitute/hellbender/tools/walkers/genotyper/afcalc/AlleleFrequencyCalculator.java (hunk 2 above)
Comment (davidbenjamin) [reply src/main/java/org/broadinstitute/hellbender/tools/walkers/genotyper/afcalc/AlleleFrequencyCalculator.java]: done

**Pull request merged**